app/server.go

```diff
@@ -3,44 +3,52 @@ package app
 import (
 	"context"
 	"fmt"
+	"io"
 	"os"
 	"os/signal"
 	"runtime"
 	"syscall"
 	"time"
 
+	"cloud.google.com/go/pubsub"
```

Comment:
@turtleDev what is the need of this dependency here?

Reply:
`publisher.NewPubSub()` doesn't create the pubsub client by itself. It leaves that responsiblity to the client code (in this case, `initPublisher()`)